tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/PromptValidationTool/DownloadPromptsTool.cs

```diff
@@ -0,0 +1,297 @@
+        private readonly Option<string> sourceRepoNameOpt = new(["--source-repo-name"], () => "azure-rest-api-specs", "Name of the source repository") { IsRequired = false };
+        private readonly Option<string> sourcePathOpt = new(["--source-path"], () => ".github/prompts", "Path in the source repository to download from") { IsRequired = false };
+        private readonly Option<string> destinationPathOpt = new(["--destination-path"], () => ".github/prompts", "Local directory path to download files to") { IsRequired = false };
+        private readonly Option<string> missingFilesOpt = new(["--missing-files"], "Comma-delimited list of file paths to download (e.g., 'file1.md,file2.md')") { IsRequired = false };
+
+        public override Command GetCommand()
+        {
+            Command command = new("download-prompts");
+            command.AddOption(sourceRepoOwnerOpt);
+            command.AddOption(sourceRepoNameOpt);
+            command.AddOption(sourcePathOpt);
+            command.AddOption(destinationPathOpt);
+            command.AddOption(missingFilesOpt);
+            command.SetHandler(async ctx => { await HandleCommand(ctx, ctx.GetCancellationToken()); });
+            return command;
+        }
+
+        public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            var sourceRepoOwner = ctx.ParseResult.GetValueForOption(sourceRepoOwnerOpt);
+            var sourceRepoName = ctx.ParseResult.GetValueForOption(sourceRepoNameOpt);
+            var sourcePath = ctx.ParseResult.GetValueForOption(sourcePathOpt);
+            var destinationPath = ctx.ParseResult.GetValueForOption(destinationPathOpt);
+            var missingFilesList = ctx.ParseResult.GetValueForOption(missingFilesOpt);
+
+            var result = await DownloadPrompts(sourceRepoOwner, sourceRepoName, sourcePath, destinationPath, missingFilesList);
+            ctx.ExitCode = ExitCode;
+            output.Output(result);
+        }
+
+        [McpServerTool, Description("Downloads files from a GitHub repository to a local directory, or from provided file paths")]
+        public async Task<DownloadResponse> DownloadPrompts(
+            string sourceRepoOwner, 
+            string sourceRepoName, 
+            string sourcePath, 
+            string destinationPath,
+            string? missingFilesList = null)
+        {
+            try
+            {
+                logger.LogInformation("Starting download from {sourceRepoOwner}/{sourceRepoName}/{sourcePath} to {destinationPath}", 
+                    sourceRepoOwner, sourceRepoName, sourcePath, destinationPath);
+
+                // Create destination directory if it doesn't exist
+                Directory.CreateDirectory(destinationPath);
+
+                // Step 1: Get the list of files to download
+                var filesToDownload = await GetFilesToDownloadAsync(sourceRepoOwner, sourceRepoName, sourcePath, missingFilesList);
```

Comment:
How do you handle the case of skipping the files if it already exists or are you force overwriting them?